webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala

```diff
@@ -418,6 +447,55 @@ class RudderXmlUserDetailsContextMapper(authConfigProvider: UserDetailListProvid
   }
 }
 
+/**
+  * A description of some properties of an authentication backend
+  * @param hasAdditionalRoles if the backend can provide additional roles than the ones of the default backend 
+  * @param overridesRoles if the backend is configured to override the roles of the default backend and cannot change them
+  * @param hasModifiablePassword if the backend is configured to allow password modification
+  */
+final case class AuthBackendProviderProperties private (
+    hasAdditionalRoles:    Boolean,
+    overridesRoles:        Boolean,
+    hasModifiablePassword: Boolean
+) {
+  def extendsRoles: Boolean = hasAdditionalRoles && !overridesRoles
+}
+
+object AuthBackendProviderProperties {
+  def apply(
+      name:                  String,
+      hasAdditionalRoles:    Boolean,
+      overridesRoles:        Boolean,
+      hasModifiablePassword: Boolean
+  ): AuthBackendProviderProperties = {
+    if (overridesRoles && !hasAdditionalRoles) {
+      // This is not consistent so we force overridesRoles to false
+      ApplicationLogger.error(
+        s"Backend provider properties for '${name}' are not consistent: backend does not enables providing roles but roles overrides is set to true. " +
+        s"Overriding roles will not be enabled."
+      )
```

Comment:
If you need to do that kind of check, then it means that you are not enforcing "Make invalid states unrepresentable", which we really want to enforce as much as possible. 

You likely want an ADT like:
```
sealed trait RoleExtension
object RoleExtension {
  case object None extends RoleExtension
  case object NoOverride extends RoleExtension
  case object WithOverride extends RoleExtention
}
```

Reply:
Plus with such an AST, it's easy to build an order relationship (for ex, add a priority index `sealed trait RoleExtension { def index: Int }` and compare on it for the priority rules, or take the max, etc)

Reply:
:+1: thanks for the suggestion, I would only need to move the existing definition from the user-management plugin to here in Rudder : https://github.com/Normation/rudder-plugins/pull/668/commits/b8cd1a9e225b694e0b57307fa494f904c5ab43cd#diff-50791aeada8e5190976dae763b438a4d990a8a3bbcf23b7e063a7c9ac11274c0R94-R101